### PR TITLE
Hide SaaS app config in the advanced settings section

### PR DESCRIPTION
**Pull request by amanda-ariyaratne**
### Purpose

This PR fixes the following issue.
- https://github.com/wso2/product-is/issues/18611

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Comment by SujanSanjula96** [comment]
Let's add the changeset

File: apps/console/src/features/applications/components/forms/advanced-configurations-form.tsx [hunk 1]
```diff
@@ -231,18 +233,23 @@ export const AdvancedConfigurationsForm: FunctionComponent<AdvancedConfiguration
...
+                        ariaLabel="Saas application"
+                        name="saas"
+                        label={ t("console:develop.features.applications.forms.advancedConfig.fields.saas.label") }
+                        required={ false }
+                        value={ config?.saas ? [ "saas" ] : [] }
+                        readOnly={ readOnly }
+                        data-testid={ `${ testId }-sass-checkbox` }
+                        data-componentid={ `${ testId }-sass-checkbox` }
+                        hint={ t("console:develop.features.applications.forms.advancedConfig.fields.saas.hint") }
+                        hidden={ !applicationConfig.advancedConfigurations.showSaaS }
```
Comment (SujanSanjula96) [inline apps/console/src/features/applications/components/forms/advanced-configurations-form.tsx:249]: we can add the legacyAuthzRuntime check here

Comment (amanda-ariyaratne) [reply apps/console/src/features/applications/components/forms/advanced-configurations-form.tsx]: Fixed with e1f4094fbb933690f99bbe2581117437e14f5879

File: apps/console/src/features/applications/components/forms/advanced-configurations-form.tsx [hunk 2]
```diff
@@ -241,7 +243,7 @@ export const AdvancedConfigurationsForm: FunctionComponent<AdvancedConfiguration
                 data-testid={ `${ testId }-sass-checkbox` }
                 data-componentid={ `${ testId }-sass-checkbox` }
                 hint={ t("console:develop.features.applications.forms.advancedConfig.fields.saas.hint") }
-                hidden={ !applicationConfig.advancedConfigurations.showSaaS }
+                hidden={ !legacyAuthzRuntime || !applicationConfig.advancedConfigurations.showSaaS }
```
Comment (brionmario) [inline apps/console/src/features/applications/components/forms/advanced-configurations-form.tsx:246]: Should we put this under `legacy mode` configs? @SujanSanjula96 ?

Comment (brionmario) [reply apps/console/src/features/applications/components/forms/advanced-configurations-form.tsx:246]: If there's no incremental migration, then it's okay to have `legacyAuthzRuntime`

Comment (SujanSanjula96) [reply apps/console/src/features/applications/components/forms/advanced-configurations-form.tsx:246]: +1 to put this as a separate config under legacy mode configs.
Reference PRs:
https://github.com/wso2/carbon-identity-framework/pull/5266
https://github.com/wso2/carbon-identity-framework/pull/5282

**Review by SujanSanjula96 — APPROVED** [review]

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/applications/components/forms/advanced-configurations-form.tsx (hunk 2 above)
Comment (amanda-ariyaratne) [reply apps/console/src/features/applications/components/forms/advanced-configurations-form.tsx]: Added new config under `legacy mode`

**Pull request merged**